apps/api/src/app/step-schemas/types/index.ts

```diff
@@ -11,4 +11,22 @@ export const customActionStepSchema = {
   additionalProperties: false,
 } as const satisfies Schema;
 
-export type SchemaOutput = (typeof schemasList)[number] | typeof customActionStepSchema;
+export const mapStepTypeToOutput = {
+  [ChannelStepEnum.SMS]: channelStepSchemas[ChannelStepEnum.SMS].output,
+  [ChannelStepEnum.EMAIL]: channelStepSchemas[ChannelStepEnum.EMAIL].output,
+  [ChannelStepEnum.PUSH]: channelStepSchemas[ChannelStepEnum.PUSH].output,
```

Comment:
Could we consider having a type file instead of the index, while keeping the index for exports? I find this approach less confusing.

Reply:
usually i refactor once it becomes messy

Reply:
done